### PR TITLE
Add husky hooks, CI tests and purge dependencies

### PR DESCRIPTION
**Pull request by Siegrift**
- [x] Setup Husky with pre-push hook - I've only added pre-push hook (because pre-commit is unusable if you do rebase) and I've only included prettier and eslint. Let me know if you prefer also `tsc` and `tests`.
- [x] Add GH actions lint/test
- [x] Make lint and tests pass
- [x] Purge and make consistent use of dependencies. Each workspace defines it's own dependencies, but I moved common dev dependencies to monorepo root. This makes it convenient because the the project doesn't need to install it and all projects will use the same version by default.

### CODE REVIEW

File: packages/api/.eslintignore [hunk 1]
```diff
@@ -0,0 +1 @@
+../../.eslintignore
```
Comment (andreogle) [inline packages/api/.eslintignore:1]: Are these files needed?

**Review by andreogle — APPROVED** [review]

File: packages/api/.eslintignore (hunk 1 above)
Comment (Siegrift) [reply packages/api/.eslintignore:1]: https://github.com/api3dao/signed-api/pull/4#discussion_r1328618447
